clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/v3client/RegionScopedV3TitusClient.java

```diff
@@ -197,9 +204,33 @@ public void terminateJob(TerminateJobRequest terminateJobRequest) {
 
   @Override
   public void terminateTasksAndShrink(TerminateTasksAndShrinkJobRequest terminateTasksAndShrinkJob) {
-    terminateTasksAndShrinkJob.getTaskIds().forEach(id ->
-      grpcBlockingStub.killTask(TaskKillRequest.newBuilder().setTaskId(id).setShrink(terminateTasksAndShrinkJob.isShrink()).build())
+    List<String> failedTasks = new ArrayList<>();
+    terminateTasksAndShrinkJob.getTaskIds().forEach(id -> {
+        try {
+          killTaskWithRetry(id, terminateTasksAndShrinkJob);
+        } catch (Exception e) {
+          failedTasks.add(id);
+          log.error("Failed to terminate and shrink titus task {}. Exception: {}", id, e);
+        }
+      }
     );
+    if (!failedTasks.isEmpty() && failedTasks.size() > 0) {
```

Comment:
if `failedTasks` isn't empty, it's going to have > 0 elements in it ... second part of this condition shouldn't be necessary?